### PR TITLE
Refactor optional dependencies, make zarr optional in pace.driver

### PR DESCRIPTION
**Pull request by mcgibbon**
## Code changes:

- pace.driver will no longer raise on import if zarr is not installed
- reduce duplication of code handling for optional imports in pace.driver and pace.util

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: driver/pace/driver/report.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import numpy as np
 
-from pace.util.mpi import MPI
+from .comm import Comm
```
Comment (jdahm) [inline driver/pace/driver/report.py:9]: Should this be `from pace.util.comm import Comm`?

Comment (mcgibbon) [reply driver/pace/driver/report.py]: Yes, updated.

**Review by jdahm — APPROVED** [review]
Looks right!

**Comment by mcgibbon** [comment]
launch jenkins

**Pull request merged**